inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,516 @@
+
+    // fleet
+    int fleet_id = -999;                                    /*!< id of fleet model object*/
+    std::vector<std::shared_ptr<fims::Fleet<Type> > > fleets; /*!< shared pointer to fleet module */
+
+    // this -> means you're referring to a class member (member of self)
+
+    // std::vector<std::shared_ptr<fims::Fleet<Type>> > surveys;
+    Population()
+    {
+      this->id = Population::id_g++;
+    }
+
+    /**
+     * @brief Initialize values. Called once at the start of model run.
+     *
+     * @param nyears number of years in the population
+     * @param nseasons number of seasons in the population
+     * @param nages number of ages in the population
+     */
+    void Initialize(int nyears, int nseasons, int nages)
+    {
+      // size all the vectors to length of nages
+      nfleets = fleets.size();
+      ages.resize(nages);
+      catch_numbers_at_age.resize(nyears * nages * nfleets);
+      mortality_F.resize(nyears * nages);
+      mortality_Z.resize(nyears * nages);
+      // proportion_mature_at_age.resize(nyears * nages);
+      proportion_mature_at_age.resize((nyears + 1) * nages);
+      weight_at_age.resize(nages);
+      catch_weight_at_age.resize(nyears * nages * nfleets);
+      unfished_numbers_at_age.resize((nyears + 1) * nages);
+      numbers_at_age.resize((nyears + 1) * nages);
+      expected_catch.resize(nyears * nfleets);
+      expected_index.resize(nyears * nfleets);
+      biomass.resize((nyears + 1));
+      unfished_spawning_biomass.resize((nyears + 1));
+      spawning_biomass.resize((nyears + 1));
+      log_naa.resize(nages);
+      log_M.resize(nyears * nages);
+      naa.resize(nages);
+
+      M.resize(nyears * nages);
+
+      for (size_t fleet_ = 0; fleet_ < this->nfleets; fleet_++)
+      {
+        this->fleets[fleet_]->Initialize(nyears, nages);   
+      }
+   
+    }
+
+    /**
+     * @brief Prepare to run the population loop. Called at each model iteration, and used
+     *  to zero out derived quantities, values that were summed, etc.
+     *
+     */
+    void Prepare()
+    {
+      std::fill(unfished_spawning_biomass.begin(), unfished_spawning_biomass.end(), 0);
+      std::fill(spawning_biomass.begin(), spawning_biomass.end(), 0);
+      std::fill(mortality_F.begin(), mortality_F.end(), 0);
+      std::fill(expected_catch.begin(), expected_catch.end(), 0);
+
+      // Transformation Section
+      for (size_t age = 0; age < this->nages; age++)
+      {
+        // for (size_t age = 0; age <= this->nages; age++) {
+        // Use fims::exp?
+        this->naa[age] = fims::exp(this->log_naa[age]);
+        for (size_t year = 0; year < this->nyears; year++)
+        {
+          int index_ya = year * this->nages + age;
+          this->M[index_ya] = fims::exp(this->log_M[index_ya]);
+        }
+      }
+      for (size_t fleet_ = 0; fleet_ < this->nfleets; fleet_++)
+      {
+        this->fleets[fleet_]->Prepare();   
```

Comment:
Is this the correct place to call fleets[fleet_]->Prepare @msupernaw ?

Reply:
As per advice from @msupernaw, removing the fleet prepare call from population.hpp